lib/libpcsc-cpp/src/SmartCard.cpp

```diff
@@ -84,15 +98,21 @@ class CardImpl
         // TODO: debug("Protocol: " + to_string(protocol()))
         try {
             DWORD size = 0;
-            std::array<BYTE, 256> feature {};
-            SCard(Control, cardHandle, DWORD(CM_IOCTL_GET_FEATURE_REQUEST), nullptr, 0U,
-                  feature.data(), DWORD(feature.size()), &size);
-            for (auto p = feature.cbegin(); DWORD(std::distance(feature.cbegin(), p)) < size;) {
-                auto tag = DRIVER_FEATURES(*p++);
-                uint32_t value = 0;
-                for (unsigned int i = 0, len = *p++; i < len; ++i)
-                    value |= uint32_t(*p++) << 8 * i;
-                features[tag] = ntohl(value);
+            std::array<BYTE, 256> buf {};
+            SCard(Control, cardHandle, DWORD(CM_IOCTL_GET_FEATURE_REQUEST), nullptr, 0U, buf.data(),
+                  DWORD(buf.size()), &size);
+            features = parseTLV<DRIVER_FEATURES>(buf, size, [](uint32_t t) { return ntohl(t); });
+
+            if (auto ioctl = features.find(FEATURE_GET_TLV_PROPERTIES); ioctl != features.cend()) {
+                SCard(Control, cardHandle, ioctl->second, nullptr, 0U, buf.data(),
+                      DWORD(buf.size()), &size);
+                auto properties = parseTLV<TLV_PROPERTIES>(buf, size, [](uint32_t t) { return t; });
```

Comment:
I haven't checked the spec and I think this is probably correct, but let me ask just in case - you are using `ntohl` with `DRIVER_FEATURES` and not using it with `TLV_PROPERTIES`. Is this as intended?